sencha-workspace/SlateTasksStudent/app/view/TaskTree.js

```diff
@@ -171,39 +195,30 @@ Ext.define('SlateTasksStudent.view.TaskTree', {
         }
     },
 
+
     // config handlers
     updateCourseSection: function(val) {
         var me = this;
 
         me.fireEvent('coursesectionchange', me, val);
     },
 
-    applyData: function(data) {
-        data.statusClasses = this.getStatusClasses();
-        return data;
-    },
-
-    updateData: function(data) {
-        this.update(data);
-    },
 
     // event handlers
     onTreeClick: function(ev, t) {
         var me = this,
             target = Ext.get(t),
-            parentEl, recordId;
+            parentEl = target.up('.slate-tasktree-item'),
+            recordId;
 
         if (target.is('.slate-tasktree-nub.is-clickable')) {
-            parentEl = target.up('.slate-tasktree-item');
             parentEl.toggleCls('is-expanded');
             me.resizeSubtasksContainer(parentEl);
             me.resizeParentContainer();
-        } else {
-            parentEl = target.up('.slate-tasktree-item');
-            if (parentEl) {
-                recordId = parentEl.dom.getAttribute('recordId');
-                me.fireEvent('itemclick', recordId);
-            }
+        } else if (parentEl) {
+            // console.log(parentEl.dom.getAttribute('data-id'));
+            recordId = parseInt(parentEl.dom.getAttribute('data-id'), 10);
+            me.fireEvent('itemclick', recordId);
```

Comment:
@BillClinton by framework convention, the first argument to any event should be the context of the event, i.e. `me.fireEvent('itemclick', me, recordId);`

Also, if an underlying UI event like `click` is in the name of the event (as opposed to something abstract and app-centric like `*select`) you should supply the UI event object as an additional parameter: `me.fireEvent('itemclick', me, recordId, ev);`